main.go

```diff
@@ -89,12 +90,50 @@ func main() {
 		Namespace:              watchNamespace,
 	})
 	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+		setupLog.Error(err, "unable to create manager")
 		os.Exit(1)
 	}
 
+	ctx := ctrl.SetupSignalHandler()
+
+	// Fetch ClientConfigs and create the clientCache
 	clientCache := clientcache.New(mgr.GetClient(), scheme)
 
+	cConfigs := k8ssandraiov1alpha1.ClientConfigList{}
+	err = mgr.GetClient().List(ctx, &cConfigs)
+	if err != nil {
+		setupLog.Error(err, "unable to fetch cluster connections")
+		os.Exit(1)
+	}
+
+	for _, cCfg := range cConfigs.Items {
+		// Create clients and add them to the client cache
+		cfg, err := clientCache.GetRestConfig(&cCfg)
```

Comment:
As I mentioned in comments elsewhere in the PR, I would to still like the ability to create multiple client connections from a single kubeconfig. I am fine with and actually prefer to address in a follow up PR. Or maybe we find it isn't necessary.

Reply:
This PR can still do that. You only need multiple ClientConfigs, but they all can point to the same Secret (which has a single kubeconfig with multiple contexts).

Reply:
excellent! 👌